### PR TITLE
[docs] Updated How to Work on Challenge Files to explain how to test individual challenges

### PR DESCRIPTION
**Pull request by RandellDawson**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Related to discussion in PR #37161

This PR adds important information on how to test individual challenges (regardless of language) instead of using `npm run test:curriculum` to make the testing faster due to not having to test all the challenges of a particular language.

### CODE REVIEW

File: docs/how-to-work-on-coding-challenges.md [hunk 1]
```diff
@@ -304,6 +304,26 @@ function myFunc() {
...
+  cd curriculum
+```
+
+3. Run the following for each challenge file for which you have added solution:
+
+```
+npm run test -- -g 'the full English title of the challenge'
+```
+    
+Once you tested each applicable challenge file using the syntax above and the test results show all tests pass, then you can create a pull request.
```
Comment (ojeytonwilliams) [inline docs/how-to-work-on-coding-challenges.md]:
```suggestion
Once you have verified that each challenge you've worked on passes the tests, please create a pull request.
```
This is a touch more concise and little bit more welcoming.  After all, if they've actually read this far, chances are they're motivated to help!

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Could we fit in something like:

> To test all challenges run `npm run test:curriculum`, to limit the tests to one challenge `<do the stuff in this PR>`

or would that make it a bit too wordy?

File: docs/how-to-work-on-coding-challenges.md [hunk 2]
```diff
@@ -304,6 +304,26 @@ function myFunc() {
...
+
+1. In the `.env` file, set the environment variable `TEST_CHALLENGES_FOR_LANGS` to the language of the challange(s) you are needing to test.  The currently accepted values are `english`, `arabic`, `chinese`, `portuguese`, `russian` and `spanish`.
+
+2.  Switch to the `curriculum` directory:
+
+```
+  cd curriculum
+```
+
+3. Run the following for each challenge file for which you have added solution:
```
[the remark this answers is not in the document]
Comment (RandellDawson) [reply docs/how-to-work-on-coding-challenges.md]: @ojeytonwilliams Actually, I think it needs to be:
> 3. Run the following for each challenge file for which you have changed any of the `testStrings` in the `Tests` section or added solutions:

If you agree, change the suggestion, so I can commit it.

[the remark this answers is not in the document]
Comment (ojeytonwilliams) [reply docs/how-to-work-on-coding-challenges.md]: Is the edit okay? I went with slightly different wording, but it's hardly a deal breaker.

**Comment by RandellDawson** [comment]
> Could we fit in something like:
>
**To test all challenges run npm run test:curriculum, to limit the tests to one challenge <do the stuff in this PR>**
>
> or would that make it a bit too wordy?

Maybe add how testing a single challenge takes much less time?

File: docs/how-to-work-on-coding-challenges.md [hunk 3]
```diff
@@ -304,6 +304,26 @@ function myFunc() {
 </details>
 ````
 
+## Testing Challenges
+
+Before you [create a pull request](how-to-open-a-pull-request.md) for your changes, you need to validate that the changes you have made do not inadvertently cause problems with the challenge.  You do this by performing the following steps:
```
Comment (ojeytonwilliams) [inline docs/how-to-work-on-coding-challenges.md]:
```suggestion
Before you [create a pull request](how-to-open-a-pull-request.md) for your changes, you need to validate that the changes you have made do not inadvertently cause problems with the challenge.  To test all challenges run `npm run test:curriculum`. To save time you can limit the tests to one challenge by performing the following steps:
```
That seems like a good way of getting across why you might want to run one test.  Reasonable?

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Comment by ojeytonwilliams** [comment]
Thanks @RandellDawson. I'll leave it for @raisedadead to take a look at, but I'm happy with it as is.
